### PR TITLE
Upgrade iTerm2 Beta to 2.9.20160102

### PR DESCRIPTION
**Pull request by robertgzr**
update version
add version substitution for the url
minor edits based on rubocop output

### CODE REVIEW

File: Casks/iterm2-beta.rb [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
-  version '2.9.20151111'
-  sha256 'fb1c566412656c4e9f5cfdc892a1ea960d0b3124479fd4c60dc65ce48ee37c2e'
+cask "iterm2-beta" do
+  version "2.9.20160102"
+  sha256 "239ee19a0dce38cd95735fb29b901d80c340eb41e9c8282109ba2d224496ee07"
 
-  url 'https://iterm2.com/downloads/beta/iTerm2-2_9_20151111.zip'
-  name 'iTerm2'
-  homepage 'https://www.iterm2.com/'
+  url "https://iterm2.com/downloads/beta/iTerm2-#{version.tr(".", "_")}.zip"
```
Comment (vitorgalvao) [inline Casks/iterm2-beta.rb]: Use `#{version.dots_to_underscores}` instead ([all methods](https://github.com/caskroom/homebrew-cask/blob/master/CONTRIBUTING.md#version-methods)).

File: Casks/iterm2-beta.rb [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
+  name "iTerm2"
```
Comment (vitorgalvao) [inline Casks/iterm2-beta.rb]: Double quotes are reserved for cases where string interpolation is needed. Please change these (and every other case) to single quotes.

**Comment by robertgzr** [comment]
Do you want me to squash the commits?

**Comment by vitorgalvao** [comment]
Yes, please.

**Pull request merged**